tests/js/click/click-mock-utils.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright (c) 2020 The Gamepad Navigator Authors
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* global gamepad */
+
+(function (fluid) {
+    "use strict";
+
+    fluid.registerNamespace("gamepad.tests.utils");
+    fluid.registerNamespace("gamepad.tests.utils.dropdown");
+    fluid.registerNamespace("gamepad.tests.utils.element");
+
+    /**
+     *
+     * Generates a mock for the unidirectional button tests for scroll.
+     *
+     * @param {Object} inputButton - The index number of the button.
+     *
+     * @return {Array} - The mock of the gamepad.
+     *
+     */
+    gamepad.tests.utils.dropdown.click = function (inputButton) {
+        var gamepadMock = null,
+            inputSpec = { buttons: {} };
+
+        // Initial count is set to 2 in all cases.
+        if (gamepad.tests.count === 1 || gamepad.tests.count === 3) {
+            inputSpec.buttons[inputButton] = 1;
+        }
+
+        // Return a list of gamepad objects (with no buttons/axes disturbed by default).
+        gamepadMock = gamepad.tests.utils.generateGamepadMock(inputSpec);
+
+        // Reduce the count on every call.
+        gamepad.tests.count--;
```

Comment:
Manipulating a global variable to track the click state seems dangerous, especially if you have multiple tests in the same file.  I also don't quite understand the pattern, i.e. why would it ever be 3?

I understand not wanting to make a component just to track the state, one idea might be to stash the number of times an element has been clicked as an attribute of the element itself.  That would help you avoid leaving key code inline in an onclick attribute as well.

Reply:
The count variable is not the number of times the button or any other element is clicked. But rather to track at which call to the polling function, we have to update the gamepad (like pressing the button at the second call to the polling function and releasing it in the next call).

Reply:
Sorry, if you are confused by the comment at [line #35](https://github.com/fluid-lab/gamepad-navigator/pull/16/files/fa1f864f9c40f02ca58b936f3c9062baf4da8925#diff-613bbd0f1614a81ad43f0c869a97f8b4R35) then please ignore it. It's incorrect and I'll fix it in my next commit.

Reply:
Yes, I kind of figured that out.  I see why you had to do this, but at a minimum the count needs to be renamed and documented better.  I still don't like making it a global variable, though.  I know you have setup and teardown, but each gamepad is manipulating the same variable.  If you could make it a member variable of the gamepad itself somehow, that would be better.